src/main/kotlin/com/pubnub/api/crypto/CryptoModule.kt

```diff
@@ -0,0 +1,110 @@
+package com.pubnub.api.crypto
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.crypto.cryptor.AesCbcCryptor
+import com.pubnub.api.crypto.cryptor.Cryptor
+import com.pubnub.api.crypto.cryptor.HeaderParser
+import com.pubnub.api.crypto.cryptor.LegacyCryptor
+import com.pubnub.api.crypto.cryptor.ParseResult
+import com.pubnub.api.crypto.data.EncryptedData
+
+class CryptoModule internal constructor(
+    internal val primaryCryptor: Cryptor,
+    internal val cryptorsForEncryptionOnly: MutableList<Cryptor> = mutableListOf(),
+    internal val headerParser: HeaderParser = HeaderParser()
+) {
+
+    init {
+        setUpDecryptorsList()
+    }
+
+    constructor(primaryCryptor: Cryptor, cryptorsForEncryptionOnly: MutableList<Cryptor> = mutableListOf()) : this(
+        primaryCryptor,
+        cryptorsForEncryptionOnly,
+        HeaderParser()
+    )
+
+    companion object {
+        fun createLegacyCryptoModule(cipherKey: String, randomIv: Boolean = true): CryptoModule {
+            return CryptoModule(
+                primaryCryptor = LegacyCryptor(cipherKey, randomIv),
+                cryptorsForEncryptionOnly = mutableListOf(AesCbcCryptor(cipherKey))
```

Comment:
```suggestion
                cryptorsForEncryptionOnly = listOf(AesCbcCryptor(cipherKey))
```

Reply:
I will change it to :
cryptorsForDecryptionOnly = listOf(LegacyCryptor(cipherKey, randomIv), AesCbcCryptor(cipherKey))